src/pair_rann.h

```diff
@@ -0,0 +1,171 @@
+/* -*- c++ -*- ----------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+/*  ----------------------------------------------------------------------
+   Contributing authors: Christopher Barrett (MSU) barrett@me.msstate.edu
+   	   	   	   	   	     Doyl Dickel (MSU) doyl@me.msstate.edu
+    ----------------------------------------------------------------------*/
+/*
+“The research described and the resulting data presented herein, unless
+otherwise noted, was funded under PE 0602784A, Project T53 "Military
+Engineering Applied Research", Task 002 under Contract No. W56HZV-17-C-0095,
+managed by the U.S. Army Combat Capabilities Development Command (CCDC) and
+the Engineer Research and Development Center (ERDC).  The work described in
+this document was conducted at CAVS, MSU.  Permission was granted by ERDC
+to publish this information. Any opinions, findings and conclusions or
+recommendations expressed in this material are those of the author(s) and
+do not necessarily reflect the views of the United States Army.​”
+
+DISTRIBUTION A. Approved for public release; distribution unlimited. OPSEC#4918
+ */
+
+#ifdef PAIR_CLASS
+
+PairStyle(rann,PairRANN)
+
+#else
+
+#ifndef LMP_PAIR_RANN
+#define LMP_PAIR_RANN
+
+#define MAXLINE 1024
+
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include "atom.h"
+#include "force.h"
+#include "comm.h"
+#include "memory.h"
+#include "neighbor.h"
+#include "neigh_list.h"
+#include "neigh_request.h"
+#include "memory.h"
+#include "error.h"
+#include "update.h"
+#include "pair.h"
+#include <map>
+#include <string>
+
+namespace LAMMPS_NS {
+
+class PairRANN : public Pair {
+ public:
+
+  //inherited functions
+  PairRANN(class LAMMPS *);
+  ~PairRANN();
+  void compute(int, int);
+  void settings(int, char **);
+  void coeff(int, char **);
+  void init_style();
+  double init_one(int, int);
+  void init_list(int , NeighList *);
+  void errorf(const char*);
+  int count_words(char *);
+  //black magic for modular fingerprints and activations
```

Comment:
this should be done differently. please see the prototype code provided elsewhere.

Reply:
I've made some changes to this which resolve the conflicts with the two build systems. I haven't done it exactly the way you suggested though. The reason why is that I would prefer that adding fingerprints/activations require as little modification of existing code as possibly. Reusing the LAMMPS style approach was a good way to do that. 

I agree that there is no reason to have LAMMPS build the style files on each compile. Instead, I suggest that developers can simply add new fingerprints/activations manually to the list of headers in rann_list_*. Changing these names to not use style_* fixes the build conflicts. Hopefully this approach is ok.